code/modules/Martial Arts/mimejutsu.dm

```diff
@@ -0,0 +1,119 @@
+#define MIMECHUCKS_COMBO "DH"
+#define MIMESMOKE_COMBO "DD"
+#define MIMEPALM_COMBO "GD"
+
+/datum/martial_art/mimejutsu
+	name = "Mimejutsu"
+	help_verb = /mob/living/carbon/human/proc/mimejutsu_help
+
+/datum/martial_art/mimejutsu/proc/check_streak(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	if(findtext(streak,MIMECHUCKS_COMBO))
+		streak = ""
+		mimeChuck(A,D)
+		return 1
+	if(findtext(streak,MIMESMOKE_COMBO))
+		streak = ""
+		mimeSmoke(A,D)
+		return 1
+	if(findtext(streak,MIMEPALM_COMBO))
+		streak = ""
+		mimePalm(A,D)
+		return 1
+	return 0
+
+/datum/martial_art/mimejutsu/proc/mimeChuck(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	if(!D.stat && !D.stunned && !D.weakened)
+		var/damage = rand(5, 8) + A.species.punchdamagelow
+		if(!damage)
```

Comment:
it could if the species dealed negative punch damage
which is probably never supposed to happen but that's a hilarious idea